### PR TITLE
Lines between rulesets

### PR DESCRIPTION
**Pull request by ravasthi**
Add `lines-between-rulesets` option.

Taking over where #436 left off; apologies that the history looks a bit messy.

@sodatea Please review. Thanks!

### CODE REVIEW

File: src/options/lines-between-rulesets.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    this.space = gonzales.createNode({type: 'space', content: this.newLines});
+    this.processBlock(ast);
+  },
+
+  processBlock(x) {
+    let that = this;
+
+    if (x.is('stylesheet')) {
+      // Check all @rules
+      this.findAtRules(x);
```
Comment (sodatea) [inline src/options/lines-between-rulesets.js]: Better name this function as `processAtRules` or some other name, as it does more than just finding at rules.

Comment (ravasthi) [inline src/options/lines-between-rulesets.js]: Will do, as well as `findRulesets`.

File: src/options/lines-between-rulesets.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+      // Check all rulesets
+      this.findRuleSets(x);
+    }
+
+    x.forEach(function(node) {
+      if (!node.is('block')) {
+        return that.processBlock(node);
+      }
+
```
Comment (sodatea) [inline src/options/lines-between-rulesets.js]: It's better to use arrow callbacks to avoid aliasing `this`.

Comment (ravasthi) [inline src/options/lines-between-rulesets.js]: Yup, will do; I'm still kind of new to ES6.

**Comment by sodatea** [comment]
And why not add sass support for this rule? It should be quite trivial to do so.

**Comment by ravasthi** [comment]
I'll take a stab at sass support and see how it goes.

**Comment by sodatea** [comment]
Fow now, sass support is buggy. Seems that gonzales-pe returns false for some `.is('space')`. Maybe it's because they are `declarationDelimiter` type rather than simple `space` nodes in sass.

**Comment by ravasthi** [comment]
Okay! I think I got it working. Please take a look when you get a chance. Thanks!

**Comment by sodatea** [comment]
Good job! I think this PR is almost good to go.
But I've got a small problem with comments:

Source:

```css
.foo {
    background: red;
}
/* comment */
.bar {
    border: 1px solid red;
}
```

After csscomb:

```css
.foo {
    background: red;
}
/* comment */

.bar {
    border: 1px solid red;
}
```

Notice the position of the comment block. Not sure if this behavior is acceptable.

**Comment by ravasthi** [comment]
I would consider that behavior unacceptable, given that it's a fair assumption the comment belongs to the block below. It was a bit tricky, but I think I got the code to correctly handle this case for one or more single-line or multi-line comments preceding a ruleset. Please have a look.

**Comment by ravasthi** [comment]
Hi @sodatea, have you had a chance to review this yet? Please let me know if there's anything further you'd like me to change. Thanks!

**Comment by sodatea** [comment]
@ravasthi Sorry, I was on vacation for the last few days and the network connection was too poor.

There are still problems with comments in your PR.

Before:
```css
.foo {background: red;}/*comment*/.bar{border: 1px solid red;}
```

After:
```css
.foo {background: red;}/*comment*/.bar{border: 1px solid red;}
```

**Comment by ravasthi** [comment]
This was an easy fix; just one small thing I wasn't checking for. Should be all working now.

**Comment by sodatea** [comment]
OK. It looks pretty good now.

Finally, just before merging this branch, it would be much appreciated if you could rebase and squash your commits to get a clean commit history.

**Comment by ravasthi** [comment]
Okay, squashed!

**Comment by sodatea** [comment]
@ravasthi Thanks!

**Comment by ravasthi** [comment]
:+1: Looking forward to the next release!

**Comment by zhe** [comment]
Looking forward to it!

**Comment by jwhitmarsh** [comment]
Is there any word on this working for Sass?

**Pull request merged**